src/widgets/ot_pulltorefresh.eliom

```diff
@@ -63,24 +66,26 @@ module Make (Conf : CONF) = struct
 
   let touchmove_handler ev _ =
     scroll_handler ();
-    Dom_html.stopPropagation ev;
-    if !dragStart >= 0
-    then
-      if !refreshFlag
-      then Dom.preventDefault ev
-      else if ev##.touches##.length = 1
-      then (
-        let target = ev##.changedTouches##item 0 in
-        Js.Optdef.iter target (fun target ->
-            distance :=
-              Float.sqrt (float_of_int (- !dragStart + target##.clientY))
-              *. scale);
-        (*move the container if and only if at the top of the document and
+    if%lwt Lwt.return (not !scrollingX)
+    then (
+      Dom_html.stopPropagation ev;
+      if !dragStart >= 0
+      then
+        if !refreshFlag
+        then Dom.preventDefault ev
+        else if ev##.touches##.length = 1
+        then (
+          let target = ev##.changedTouches##item 0 in
+          Js.Optdef.iter target (fun target ->
+              let dY = - !dragStart + target##.clientY in
+              distance := Float.sqrt (float_of_int dY) *. scale;
+              scrollingX := abs (!scrollXStart - target##.clientX) > abs dY);
+          (*move the container if and only if at the top of the document and
             the page is scrolled down*)
-        if !top && !distance > 0.
-        then touchmove_handler_ ev
-        else joinRefreshFlag := false);
-    Lwt.return_unit
+          if !top && !distance > 0.
```

Comment:
&& not !scrollingX ?

Reply:
already checked at the top, no need to do that twice.